syntax = "proto3";

option go_package = "bonfire/transcode";

import "google/protobuf/empty.proto";

message StartAudioSlotRequest
{
  string room_id = 1;
  string sink_hostname = 2;
  uint32 sink_port = 3;
  bool listener = 4;
}

message AudioSlotInfo
{
  string id = 1;
  string hostname = 2;
  uint32 port = 3;
}

message StopAudioSlotRequest
{
  string slot_id = 1;
}

service AudioMixer
{
  rpc StartAudioSlot(StartAudioSlotRequest) returns (AudioSlotInfo);
  rpc StopAudioSlot(StopAudioSlotRequest) returns (google.protobuf.Empty);
}